src/lsptoolshost/utils/combineDocumentSelectors.ts

```diff
@@ -4,7 +4,10 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as vscode from 'vscode';
+import { DocumentSelector } from 'vscode-languageserver-protocol/lib/common/protocol';
 
-export function combineDocumentSelectors(...selectors: vscode.DocumentSelector[]): vscode.DocumentSelector {
-    return selectors.reduce<(string | vscode.DocumentFilter)[]>((acc, selector) => acc.concat(selector), []);
+export function combineDocumentSelectors(
+    ...selectors: (DocumentSelector | vscode.DocumentSelector)[]
+): vscode.DocumentSelector {
+    return selectors.reduce<(string | vscode.DocumentFilter)[]>((acc, selector) => acc.concat(<any>selector), []);
```

Comment:
DocumentSelector types do not quite match up anymore. I expect this will be resolved in a future "@types/vscode" package update.

Reply:
Shouldn't we be using the vscode document selector types here?  And not the LSP versions?

Reply:
we weren't previously but that doesn't mean that we can't. Will look at it.

Reply:
Long story short, we need to use both types of DocumentSelector. ClientLanguageOptions requires the LSP type and various calls require the vscode type. When vscode moves to the newer vscode-languageserver package, I expect these to become equivalent again.